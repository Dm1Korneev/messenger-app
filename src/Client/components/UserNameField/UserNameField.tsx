import { TextField } from 'formik-material-ui';

type Props = Omit<
  React.ComponentPropsWithoutRef<typeof TextField>,
  'margin' | 'label' | 'fullWidth' | 'autoComplete'>

export const UserNameField = (props: Props) => (
  <TextField
    autoComplete="name"
    fullWidth
    label="Name *"
    margin="normal"
    {...props}
  />
);
