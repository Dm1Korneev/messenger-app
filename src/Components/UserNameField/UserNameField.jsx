import React from 'react';
import { TextValidator } from 'react-material-ui-form-validator';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';

function UserNameField(props) {
  const { value, onChange } = props;

  return (
    <TextValidator
      margin="normal"
      label="Name *"
      fullWidth
      onChange={onChange}
      name="name"
      id="name"
      autoComplete="name"
      color="primary"
      value={value}
      validators={['required']}
      errorMessages={['this field is required']}
    />
  );
}

UserNameField.defaultProps = {
  value: undefined,
};
UserNameField.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

const styles = () => ({});

export default withStyles(styles)(UserNameField);
