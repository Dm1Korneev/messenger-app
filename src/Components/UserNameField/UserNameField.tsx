import { TextField } from 'formik-material-ui';

type Props = Omit<
  React.ComponentPropsWithoutRef<typeof TextField>,
  'margin' | 'label' | 'fullWidth' | 'autoComplete'>

export const UserNameField = (props: Props) => (
  <TextField
    margin="normal"
    label="Name *"
    fullWidth
    autoComplete="name"
    {...props}
  />
);
