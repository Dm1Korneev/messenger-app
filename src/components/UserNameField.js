import React from "react";
import { withStyles } from "@material-ui/core/styles";
import { TextValidator } from "react-material-ui-form-validator";

function UserNameField(props) {
  const { value, onChange } = props;

  return (
    <TextValidator
      margin="normal"
      label="Name *"
      fullWidth
      onChange={onChange}
      name="name"
      id="name"
      autoComplete="name"
      color="primary"
      value={value}
      validators={["required"]}
      errorMessages={["this field is required"]}
    />
  );
}

const styles = theme => ({});

export default withStyles(styles)(UserNameField);
