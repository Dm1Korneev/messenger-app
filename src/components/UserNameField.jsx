import React from 'react';
import { TextValidator } from 'react-material-ui-form-validator';
import PropTypes from 'prop-types';

// material-ui
import withStyles from '@material-ui/core/styles/withStyles';

function UserNameField(props) {
  const { value, onChange } = props;

  return (
    <TextValidator
      margin="normal"
      label="Name *"
      fullWidth
      onChange={onChange}
      name="name"
      id="name"
      autoComplete="name"
      color="primary"
      value={value}
      validators={['required']}
      errorMessages={['this field is required']}
    />
  );
}

UserNameField.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

const styles = (theme) => ({});

export default withStyles(styles)(UserNameField);
