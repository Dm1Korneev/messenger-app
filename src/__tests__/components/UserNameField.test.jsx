import React from 'react';

import UserNameField from 'Components/UserNameField';

jest.mock('react-material-ui-form-validator', () => ({
  TextValidator: global.mockComponent('TextValidator'),
}));

describe('render UserPasswordField component', () => {
  let wrapper;

  const props = {
    value: 'TEST',
    onChange: jest.fn(),
  };

  beforeAll(() => {
    wrapper = global.mount(
      <UserNameField {...props} />,
    );
  });

  test('component is render', () => {
    expect(wrapper.find('UserNameField').length).toBe(1);
  });

  test('TextValidator subcomponent is render', () => {
    wrapper.update();
    expect(wrapper.find({ originalcomponent: 'TextValidator' }).length).toBe(1);
  });
});

describe('snapshot-test UserNameField component', () => {
  test('Renders correct properties', () => {
    const props = {
      value: 'TEST',
      onChange: jest.fn(),
    };

    global.mountExpect(
      <UserNameField {...props} />,
    ).toMatchSnapshot();
  });
});
