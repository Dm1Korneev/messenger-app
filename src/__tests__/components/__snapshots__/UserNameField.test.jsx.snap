// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot-test UserNameField component Renders correct properties 1`] = `
<WithStyles(UserNameField)
  onChange={[MockFunction]}
  value="TEST"
>
  <UserNameField
    classes={Object {}}
    onChange={[MockFunction]}
    value="TEST"
  >
    <Component
      autoComplete="name"
      color="primary"
      errorMessages={
        Array [
          "this field is required",
        ]
      }
      fullWidth={true}
      id="name"
      label="Name *"
      margin="normal"
      name="name"
      onChange={[MockFunction]}
      validators={
        Array [
          "required",
        ]
      }
      value="TEST"
    >
      <div
        originalcomponent="TextValidator"
        testprops={
          Object {
            "autoComplete": "name",
            "color": "primary",
            "errorMessages": Array [
              "this field is required",
            ],
            "fullWidth": true,
            "id": "name",
            "label": "Name *",
            "margin": "normal",
            "name": "name",
            "onChange": [MockFunction],
            "validators": Array [
              "required",
            ],
            "value": "TEST",
          }
        }
      />
    </Component>
  </UserNameField>
</WithStyles(UserNameField)>
`;
